src/test/java/com/redhat/devtools/lsp4ij/features/completion/CompletionItemComparatorTest.java

```diff
@@ -16,10 +16,11 @@
 import java.util.ArrayList;
 import java.util.List;
 
-import static org.junit.jupiter.api.Assertions.*;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
 
 public class CompletionItemComparatorTest {
-    private final CompletionItemComparator comparator = new CompletionItemComparator();
+    private final CompletionItemComparator comparator = new CompletionItemComparator(null, null, false);
```

Comment:
I'll add more test cases here to confirm proper behavior of the new context-aware features of this comparator.